### DISCUSSION TITLE
creating an entry in Table A from an entry in table B

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6841] [Help]
Im not sure if this is possible and a bit scared to try in case it opens a can of worms as my app is currently live.

My original table had a dropdown `options picker` with the options

- job in progress
- invoiced
- sourcing
- quoted

However `Quoted and sourcing` are intended to be kept separate from the other 2 options when displayed on the app screen. However, due to the filtering not being great, even when I chose the filters shown in the pic (Igonre archived for now)

![image](https://user-images.githubusercontent.com/75368133/180233669-39b300d7-cdbb-4a91-9d7c-a9b38322caaf.png)

It still shows entries which have used these options...

As a result I have opted to make a new table for my sourcing and quoting.
I have created it with identical schema as the initial table.

What I am wondering is... Once I have quoted my customer and they accept the job... Is it possible to have a button in the `sourcing/:id` page where I can turn the quote into a live job and have it populate the `livejob` table?

The thing that scares me slightly with this, isnt so much the data... It is the `relationship` columns. I feel as though there could be some issues as the relationships in both columns would use different `Foreign Keys`

Has this been done before? Can it work?

**Comment by R2bEEaton** [comment]
This will not be so much an answer, but instead a place you can hopefully start...

### A note about editing live apps ###
When you go to edit a live app, Budibase does make a development environment that you can revert at any time before publishing. However, I find that I often need to publish an app if I want to fully test it. So I'd recommend exporting your app and reimporting it under a different name so you can publish and mess around with it as much as you want without affecting the live version.

### Before complicating things... ###
Before trying to move rows between tables, potentially breaking relationships, I'd recommend trying to keep Sourcing and Quoting JobStatuses in the same table. If your Budibase version is out of date, make sure to update it to see if filtering works like you wanted it to in the screenshot. Filtering was broken in some places not too long ago, so this may be related. If it doesn't work still or something else goes wrong, you can try filtering in other ways like hiding elements inside a repeater based on certain conditions. I just want to emphasize that keeping these in the same table will be better in the long run for ease of use and more. It may be more effort now to get it back to the same table, so use your knowledge of your specific use case to decide if the time investment is worth it.

### A potential solution ###
If you absolutely need to have these in separate tables, fortunately there is something you can do. On buttons, you can add an action to "Duplicate Row" which sounds exactly like what you want. Read more about it here: https://docs.budibase.com/docs/actions#duplicate-row (there's actually not much here). The general idea of it is that you can copy a row from a data source into another table as long as the table columns match. You can also override the `JobStatus` field to be the desired status, like `Invoiced`. After this "Duplicate Row" action, simply add a "Delete Row" action to get rid of the initial row which is no longer needed.

I hope this helps you get closer to a good solution!

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

What you're describing seems to be a bug. Would you be able to create an issue with reproduction steps?

Alternatively you can send me your app export so I can investigate, as the filter should work.
